src/app/view/loadUtils.ts

```diff
@@ -31,15 +32,45 @@ export default class LoadUtils {
     static loadFromScratch(json: ScratchInterchangeV1): [Device, Array<RenderLayer>] {
         const newDevice: Device = LoadUtils.loadDeviceFromInterchangeV1(json);
         let newRenderLayers: Array<RenderLayer> = [];
-        for (let i = 0; i < json.renderLayers.length; i++) {
-            newRenderLayers.push(LoadUtils.loadRenderLayerFromInterchangeV1(json.renderLayers[i], newDevice));
+        if (json.renderLayers) {
+            for (let i = 0; i < json.renderLayers.length; i++) {
+                newRenderLayers.push(LoadUtils.loadRenderLayerFromInterchangeV1(json.renderLayers[i], newDevice));
+            }
+        } else if (Object.prototype.hasOwnProperty.call(json, "layers")) {
+            for (let i = 0; i < json.layers.length; i++) {
+                newRenderLayers.push(LoadUtils.generateRenderLayerFromLayerInterchangeV1(json.layers[i], newDevice));
+            }
+        } else {
+            newRenderLayers.push(new RenderLayer("flow", newDevice.layers[0], LogicalLayerType.FLOW));
+            newRenderLayers.push(new RenderLayer("control", newDevice.layers[1], LogicalLayerType.CONTROL));
+            newRenderLayers.push(new RenderLayer("integration", newDevice.layers[0], LogicalLayerType.INTEGRATION));
+        }
+
+        // Ensures that there are three layers per group
+        let layerGroups: Map<string, number> = new Map();
+        for (let i = 0; i < newDevice.layers.length; i++) {
+            if (layerGroups.has(newDevice.layers[i].group)) {
+                const currentVal = layerGroups.get(newDevice.layers[i].group);
+                if (currentVal) layerGroups.set(newDevice.layers[i].group, currentVal + 1);
+            } else {
+                layerGroups.set(newDevice.layers[i].group, 1);
+            }
         }
+        layerGroups.forEach((value, key) => {
+            const keyVal = parseInt(key, 10);
+            if (value == 2) {
+                newDevice.addLayerAtIndex(new Layer({}, "integration", LogicalLayerType.INTEGRATION, key, newDevice), keyVal * 3 + 2);
+                newRenderLayers.splice(keyVal * 3 + 2, 0, new RenderLayer("integration", newDevice.layers[0], LogicalLayerType.INTEGRATION));
```

Comment:
We should generate a unique name for all the layers, otherwise this will become problematic later on